crates/sui-sdk/src/transaction_builder.rs

```diff
@@ -100,6 +101,32 @@ impl TransactionBuilder {
         ))
     }
 
+    pub async fn pay_sui(
+        &self,
+        signer: SuiAddress,
+        input_coins: Vec<ObjectID>,
+        recipients: Vec<SuiAddress>,
+        amounts: Vec<u64>,
+        gas_budget: u64,
+    ) -> anyhow::Result<TransactionData> {
+        // use the first input coin as the gas payment object, estimated gas cost will be added to
+        // this coin in transaction execution and will be deducted upon gas payment.
+        fp_ensure!(!input_coins.is_empty(), SuiError::EmptyInputCoins.into());
+        let gas_object = self.get_object_ref(input_coins[0]).await?;
```

Comment:
pick the first input coin as the gas payment object, this coin object is guaranteed to exist after `pay()`, bc 
1. `input_coins` is not empty as it is checked both here and in `pay()`
2.  `recipients` is not empty, which is checked in `pay()`, as a result, this coin will be transferred to the first recipient on L432 in execution_engine.rs.


on the other hand, if we follow the original pay(), it is possible that all `input_coins` will be deleted after `pay()`, thus here there is no coin we can use as `gas_object`.